core/src/main/scala/kafka/server/DynamicConfig.scala

```diff
@@ -0,0 +1,86 @@
+  * the License.  You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.server
+
+import java.util.Properties
+import org.apache.kafka.common.config.ConfigDef
+import org.apache.kafka.common.config.ConfigDef.Importance._
+import org.apache.kafka.common.config.ConfigDef.Range._
+import org.apache.kafka.common.config.ConfigDef.Type._
+import scala.collection.JavaConverters._
+
+/**
+  * Class used to hold dynamic configs. These are configs which have no physical manifestation in the server.properties
+  * and can only be set dynamically.
+  */
+object DynamicConfig {
+
+  object Broker {
+    //Properties
+    val ThrottledReplicationRateLimitProp = "replication.quota.throttled.rate"
+
+    //Defaults
+    val DefaultThrottledReplicationRateLimit = ReplicationQuotaManagerConfig.QuotaBytesPerSecondDefault
+
+    //Documentation
+    val ThrottledReplicationRateLimitDoc = "A long representing the upper bound (bytes/sec) on replication traffic for replicas enumerated in the " +
+      s"property $ThrottledReplicationRateLimitProp. This property can be only set dynamically via the config command etc. The minimum value is 1MB/s."
+
+    //Definitions
+    private val brokerConfigDef = new ConfigDef()
+      //round minimum value down, to make it easier for users.
+      .define(ThrottledReplicationRateLimitProp, LONG, DefaultThrottledReplicationRateLimit, atLeast(1000 * 1000), MEDIUM, ThrottledReplicationRateLimitDoc)
+
+    def names = brokerConfigDef.names
+
+    def validateNames(props: Properties) = DynamicConfig.validate(brokerConfigDef, props)
+  }
+
+  object Client {
+    //Properties
+    val ProducerByteRateOverrideProp = "producer_byte_rate"
+    val ConsumerByteRateOverrideProp = "consumer_byte_rate"
+
+    //Defaults
+    val DefaultProducerOverride = ClientQuotaManagerConfig.QuotaBytesPerSecondDefault
+    val DefaultConsumerOverride = ClientQuotaManagerConfig.QuotaBytesPerSecondDefault
+
+    //Documentation
+    val ProducerOverrideDoc = "A rate representing the upper bound (bytes/sec) for producer traffic."
+    val ConsumerOverrideDoc = "A rate representing the upper bound (bytes/sec) for consumer traffic."
+
+    //Definitions
+    val clientConfigs = new ConfigDef()
+      .define(ProducerByteRateOverrideProp, LONG, DefaultProducerOverride, MEDIUM, ProducerOverrideDoc)
+      .define(ConsumerByteRateOverrideProp, LONG, DefaultConsumerOverride, MEDIUM, ConsumerOverrideDoc)
+
+    def names = clientConfigs.names
+
+    def validateNames(props: Properties) = DynamicConfig.validate(clientConfigs, props)
```

Comment:
This and other similar methods should be called `validate` now since they do more than name validation now right?